nixos/modules/services/databases/postgresql.nix

```diff
@@ -597,12 +603,26 @@ in
                   clauseSqlStatements = attrValues (mapAttrs (n: v: if v then n else "no${n}") filteredClauses);
 
                   userClauses = ''$PSQL -tAc 'ALTER ROLE "${user.name}" ${concatStringsSep " " clauseSqlStatements}' '';
+
+                  ensureUserPasswords = lib.optionalString
+                    (user.ensurePasswordFile != null)
+                    ''
+                      $PSQL -tA <<'EOF'
+                        DO $$
+                        DECLARE password TEXT;
+                        BEGIN
+                          password := trim(both from replace(pg_read_file('${user.ensurePasswordFile}'), E'\n', '''));
+                          EXECUTE format('ALTER ROLE ${user.name} WITH PASSWORD '''%s''';', password);
```

Comment:
Do you know what happens here if the file is not readable or missing? Is that just ignored? Does the script abort?

Reply:
The unit fails to start.

Reply:
Example from `journalctl -efu postgresql`/`systemctl status postgresql`:

```
Dez 16 22:11:25 marcel-nix postgres[758943]: [758943] ERROR:  could not open file "/run/secrets/test" for reading: Permission denied
Dez 16 22:11:25 marcel-nix postgres[758943]: [758943] CONTEXT:  PL/pgSQL function inline_code_block line 4 at assignment
Dez 16 22:11:25 marcel-nix postgres[758943]: [758943] STATEMENT:  DO $$
Dez 16 22:11:25 marcel-nix postgres[758943]:           DECLARE password TEXT;
Dez 16 22:11:25 marcel-nix postgres[758943]:           BEGIN
Dez 16 22:11:25 marcel-nix postgres[758943]:             password := trim(both from replace(pg_read_file('/run/secrets/test'), E'\n', ''));
Dez 16 22:11:25 marcel-nix postgres[758943]:             EXECUTE format('ALTER ROLE test WITH PASSWORD ''%s'';', password);
Dez 16 22:11:25 marcel-nix postgres[758943]:           END $$;
Dez 16 22:11:25 marcel-nix postgresql-post-start[758942]: ERROR:  could not open file "/run/secrets/test" for reading: Permission denied
Dez 16 22:11:25 marcel-nix postgresql-post-start[758942]: CONTEXT:  PL/pgSQL function inline_code_block line 4 at assignment
```